### ISSUE TITLE
Is W1514 too pedantic?

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#4888]
### Bug description

```python
with open('/etc/passwd', 'r') as f:
    pass
```

produces diagnostic W1514. This seems excessive given that the `encoding` argument has a default (None), and it's perfectly reasonable to use the default.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:2: [W1514(unspecified-encoding), ] Using open without explicitly specifying an encoding
```

### Expected behavior

No diagnostic.

### Pylint version

```shell
pylint 2.10.1
astroid 2.7.2
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
```

**Comment by cdce8p** [comment]
The issue is that the default encoding can vary between operating systems. It's recommended to explicitly set the encoding.
See https://www.python.org/dev/peps/pep-0597/ for further details.

Python 3.10 will add the optional `EncodingWarning` which does nearly the same as this check.

**Comment by Pierre-Sassoulas** [comment]
To add to what @cdce8p said, you can disable the check if you're not aiming for cross platform compatible code.
